### PR TITLE
Add specification for STO, BrakeTest and OperationMode.

### PR DESCRIPTION
**Pull request by hslusarek**

### CODE REVIEW

File: prbt_hardware_support/spec.dox [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+ * @subsubsection sto_stop1 Safe torque off (STO) and  Stop 1
+ *
+ * @definereq{Stop1_Trigger}
+ *  - A Stop 1 is triggered, if at least one of the following cases is true:
+ *    - the state of the STO changes from TRUE -> FALSE,
+ *    - connection to safety controller is lost,
+ *    - the messages informing about the STO (coming from the
+ *      safety controller) is incomplete or in any way corrupted,
+ *    - the communication protocol used by the safety controller
+ *      does not fulfill the expected specification.
```
Comment (agutenkunst) [inline prbt_hardware_support/spec.dox]: Version?

File: prbt_hardware_support/spec.dox [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+ *  @brief System Specification
+ *
+ *  @subsection ISO-10218-1
+ *
+ *
+ *
+ * @subsubsection sto_stop1 Safe torque off (STO) and  Stop 1
+ *
+ * @definereq{Stop1_Trigger}
+ *  - A Stop 1 is triggered, if at least one of the following cases is true:
```
Comment (agutenkunst) [inline prbt_hardware_support/spec.dox:34]: We had trouble before with the meaning behind STO = false/true. Should we try to establish a better convention here?

File: prbt_hardware_support/spec.dox [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+ *    - the messages informing about the STO (coming from the
+ *      safety controller) is incomplete or in any way corrupted,
+ *    - the communication protocol used by the safety controller
+ *      does not fulfill the expected specification.
+ *
+ * @definereq{No_new_commands_during_STO_false}
+ *  - As long as the STO is FALSE, no new motions commands are allowed.
+ *
+ * @definereq{Hold_driver_if_STO_false}
+ *  - If the STO changes from TRUE -> FALSE the driver is halt.
```
Comment (agutenkunst) [inline prbt_hardware_support/spec.dox]: *halted?

File: prbt_hardware_support/spec.dox [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+ *
+ * @definereq{Recover_driver_after_STO_false}
+ *  - If the STO changes from FALSE -> TRUE, the driver is recovered.
```
Comment (agutenkunst) [inline prbt_hardware_support/spec.dox:50]: Should this also state the unholding?

File: prbt_hardware_support/spec.dox [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+ * - The system provides a mechanism which allows to execute a brake test.
+ * - The brake test is performed for all axes.
+ * - The brake test mechanism returns a status stating if the
+ *    brake test was successful xor what error occured.
+ * - At the beginning of the brake test it is checked if the robot is
+ *    currently moving. In case the robot is moving, the brake test is not
+ *    performed and an error is retured.
+ * - As long as the brake test is performed no motion is allowed. Commands
+ *    send during the execution of the brake test are rejected.
+ * - Brake tests are only performed for drives which define that they require
```
Comment (agutenkunst) [inline prbt_hardware_support/spec.dox:96]: I know you want to keep this high level but maybe a hint to the config file?

**Review by agutenkunst — CHANGES_REQUESTED** [review]

File: prbt_hardware_support/spec.dox (hunk 1 above)
Comment (hslusarek) [reply prbt_hardware_support/spec.dox]: Done.

File: prbt_hardware_support/spec.dox (hunk 3 above)
Comment (hslusarek) [reply prbt_hardware_support/spec.dox]: Done.

File: prbt_hardware_support/spec.dox (hunk 5 above)
Comment (hslusarek) [reply prbt_hardware_support/spec.dox:96]: Done.

File: prbt_hardware_support/spec.dox (hunk 4 above)
Comment (hslusarek) [reply prbt_hardware_support/spec.dox:50]: Indirectly covered requirement "No_new_commands_during_STO_false"

File: prbt_hardware_support/spec.dox (hunk 2 above)
Comment (hslusarek) [reply prbt_hardware_support/spec.dox:34]: Done.

**Pull request merged**